helm/daq-config-server/templates/deployment.yaml

```
@@ -53,14 +53,19 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
-          {{- with .Values.volumeMounts }}
           volumeMounts:
-            {{- toYaml . | nindent 12 }}
-          {{- end }}
+          - name: config-volume
+            mountPath: "/etc/config"
+            subPath: config.yaml
       {{- with .Values.volumes }}
       volumes:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+        - name: config-volume
+          projected:
+            sources:
+            - configMap:
+                name: {{ include "daq-config-server.fullname" . }}-config
```

```suggestion
      volumes:
      {{- with .Values.volumes }}
        {{- toYaml . | nindent 8 }}
      {{- end }}
        - name: config-volume
          projected:
            sources:
            - configMap:
                name: {{ include "daq-config-server.fullname" . }}-config
```

As is you only get your config-volume if any other volume is defined in values.yaml, the `with` block is ignored if the value it is with is Falsey (e.g. empty)